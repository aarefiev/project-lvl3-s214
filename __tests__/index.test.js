import 'js-polyfills/dom';
import 'js-polyfills/url';
import 'js-polyfills/html';

import fs from 'fs';
import path from 'path';
import timer from 'timer-promise';
import nock from 'nock';
import keycode from 'keycode';
import { html } from 'js-beautify';

import run from '../src';

const host = 'http://localhost';
const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML);

const pressKey = (key, el = document.body, value = key) => {
  const keyCode = keycode(key);
  const e = new KeyboardEvent('input', { keyCode });
  el.value = value; // eslint-disable-line
  el.dispatchEvent(e);
};

const submitForm = (el = document.body) => {
  const e = new Event('submit');
  el.dispatchEvent(e);
};

beforeAll(async () => {
  jsdom.reconfigure({ url: host }); // eslint-disable-line

  const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html')).toString();

  document.documentElement.innerHTML = initHtml;
  nock.disableNetConnect();
  run();
});

test('application', async () => {
  const input = document.getElementById('rss_input_url');
  const form = document.getElementById('rss_form');
  const feedURL = 'http://lorem-rss.herokuapp.com/feed';
  const feedInXML = fs.readFileSync(path.join(fixturesPath, 'feed.xml'), 'utf8')
    .toString();

  expect(getTree()).toMatchSnapshot();

  pressKey('h', input);
  await timer.start(100);
  expect(getTree()).toMatchSnapshot();

  pressKey(feedURL, input);
  await timer.start(100);
  expect(getTree()).toMatchSnapshot();

  nock(host)
    .get('/api/rss')
    .query({ url: feedURL })
    .reply(200, feedInXML);

  submitForm(form);
  await timer.start(100);
  expect(getTree()).toMatchSnapshot();
});
