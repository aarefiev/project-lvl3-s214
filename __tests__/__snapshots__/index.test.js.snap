// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div id=\\"application\\">
    <div class=\\"container\\">
        <div class=\\"row\\">
            <div class=\\"col-12\\">
                <div class=\\"d-flex justify-content-center\\">
                    <div class=\\"jumbotron col-md-6\\">
                        <form id=\\"rss_form\\">
                            <div class=\\"form-group\\">
                                <label for=\\"rss_input_url\\">URL адрес</label>
                                <input type=\\"text\\" value=\\"\\" placeholder=\\"Введите URL\\" class=\\"form-control\\" id=\\"rss_input_url\\">
                            </div>
                            <button type=\\"submit\\" class=\\"btn btn-primary\\">Добавить</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <!-- Modal -->
    <div class=\\"modal fade\\" id=\\"rss_modal\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-hidden=\\"true\\">
        <div class=\\"modal-dialog modal-dialog-centered\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\"></h5>
                    <button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
                <span aria-hidden=\\"true\\">×</span>
              </button>
                </div>
                <div class=\\"modal-body\\"></div>
                <div class=\\"modal-footer\\">
                    <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button>
                </div>
            </div>
        </div>
    </div>
</div>"
`;

exports[`application 2`] = `
"<div id=\\"application\\">
    <div class=\\"container\\">
        <div class=\\"row\\">
            <div class=\\"col-12\\">
                <div class=\\"d-flex justify-content-center\\">
                    <div class=\\"jumbotron col-md-6\\">
                        <form id=\\"rss_form\\">
                            <div class=\\"form-group\\">
                                <label for=\\"rss_input_url\\">URL адрес</label>
                                <input type=\\"text\\" value=\\"h\\" placeholder=\\"Введите URL\\" class=\\"form-control is-invalid\\" id=\\"rss_input_url\\">
                            </div>
                            <button type=\\"submit\\" class=\\"btn btn-primary\\">Добавить</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <!-- Modal -->
    <div class=\\"modal fade\\" id=\\"rss_modal\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-hidden=\\"true\\">
        <div class=\\"modal-dialog modal-dialog-centered\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\"></h5>
                    <button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
                <span aria-hidden=\\"true\\">×</span>
              </button>
                </div>
                <div class=\\"modal-body\\"></div>
                <div class=\\"modal-footer\\">
                    <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button>
                </div>
            </div>
        </div>
    </div>
</div>"
`;

exports[`application 3`] = `
"<div id=\\"application\\">
    <div class=\\"container\\">
        <div class=\\"row\\">
            <div class=\\"col-12\\">
                <div class=\\"d-flex justify-content-center\\">
                    <div class=\\"jumbotron col-md-6\\">
                        <form id=\\"rss_form\\">
                            <div class=\\"form-group\\">
                                <label for=\\"rss_input_url\\">URL адрес</label>
                                <input type=\\"text\\" value=\\"http://lorem-rss.herokuapp.com/feed\\" placeholder=\\"Введите URL\\" class=\\"form-control is-valid\\" id=\\"rss_input_url\\">
                            </div>
                            <button type=\\"submit\\" class=\\"btn btn-primary\\">Добавить</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <!-- Modal -->
    <div class=\\"modal fade\\" id=\\"rss_modal\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-hidden=\\"true\\">
        <div class=\\"modal-dialog modal-dialog-centered\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\"></h5>
                    <button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
                <span aria-hidden=\\"true\\">×</span>
              </button>
                </div>
                <div class=\\"modal-body\\"></div>
                <div class=\\"modal-footer\\">
                    <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button>
                </div>
            </div>
        </div>
    </div>
</div>"
`;

exports[`application 4`] = `
"<div id=\\"application\\">
    <div class=\\"container\\">
        <div class=\\"row\\">
            <div class=\\"col-12\\">
                <div class=\\"d-flex justify-content-center\\">
                    <div class=\\"jumbotron col-md-6\\">
                        <form id=\\"rss_form\\">
                            <div class=\\"form-group\\">
                                <label for=\\"rss_input_url\\">URL адрес</label>
                                <input type=\\"text\\" value=\\"http://lorem-rss.herokuapp.com/feed\\" placeholder=\\"Введите URL\\" class=\\"form-control is-valid\\" id=\\"rss_input_url\\">
                            </div>
                            <button type=\\"submit\\" class=\\"btn btn-primary\\">Добавить</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <!-- Modal -->
    <div class=\\"modal fade\\" id=\\"rss_modal\\" tabindex=\\"-1\\" role=\\"dialog\\" aria-hidden=\\"true\\">
        <div class=\\"modal-dialog modal-dialog-centered\\" role=\\"document\\">
            <div class=\\"modal-content\\">
                <div class=\\"modal-header\\">
                    <h5 class=\\"modal-title\\"></h5>
                    <button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\">
                <span aria-hidden=\\"true\\">×</span>
              </button>
                </div>
                <div class=\\"modal-body\\"></div>
                <div class=\\"modal-footer\\">
                    <button type=\\"button\\" class=\\"btn btn-secondary\\" data-dismiss=\\"modal\\">Close</button>
                </div>
            </div>
        </div>
    </div>
</div>"
`;
